lib/ruby_smb/smb2/packet/compression_transform_header.rb

```diff
@@ -11,6 +11,7 @@ class CompressionTransformHeader < BinData::Record
         uint16           :compression_algorithm,            label: 'Compression Algorithm'
         uint16           :flags,                            label: 'Flags'
         uint32           :offset,                           label: 'Offset / Length'
+        array            :compressed_data,                  label: 'Compressed Data', type: :uint8, read_until: :eof
       end
```

Comment:
Not a blocker, just a note here. It looks like the `COMPRESSION_TRANSFORM_HEADER` protocol definition has been updated and is now `SMB2_COMPRESSION_TRANSFORM_HEADER_CHAINED`and `SMB2_COMPRESSION_TRANSFORM_HEADER_UNCHAINED` (see https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/1d435f21-9a21-4f4c-828e-624a176cf2a0).

It is likely out of scope for this PR, but it is something we will need to update when implementing compression capability in RubySMB client.

Reply:
Wow yeah literally updated just a couple of days ago. Good catch! We can add a note that this structure is here for historical reasons because there was a point in time where it was defined as is.

Reply:
Absolutely, a note would be certainly useful. Thanks!